site/src/pages/CreateWorkspacePage/useWorkspaceDuplication.test.tsx

```diff
@@ -75,25 +74,29 @@ describe(`${useWorkspaceDuplication.name}`, () => {
       M.MockWorkspaceBuildParameter5,
     ];
 
+    const { result, getLocationSnapshot } = await render(MockWorkspace);
+    await performNavigation(result, getLocationSnapshot);
+
+    const { search } = getLocationSnapshot();
     for (const { name, value } of mockBuildParams) {
       const key = `param.${name}`;
-      expect(parsedParams.get(key)).toEqual(value);
+      expect(search.get(key)).toEqual(value);
     }
   });
 
   test("Navigating appends other necessary metadata to the search params", async () => {
-    const { result, router } = await render(MockWorkspace);
-    await performNavigation(result, router);
-
-    const parsedParams = new URLSearchParams(router.state.location.search);
-    const extraMetadataEntries = [
+    const extraMetadataEntries: readonly [string, string][] = [
```

Comment:
why the switch from `as const`?

Reply:
I felt like it would surface any accidental errors in a better way, and that it does a better job of communicating intent

With the previous code, I could do something like this:
```tsx
 const extraMetadataEntries = [
      ["mode", "duplicate"],
      ["name", `${MockWorkspace.name}-copy`],
      ["version", MockWorkspace.template_active_version_id],

      function obviouslyWrong() {},
      null,
      ["forgotToAddTheSecondTupleElement"]
] as const;
```
And this would be valid still – the underlying type would get broadened. So while the type mismatch would get caught down the line, redefining the type to be an array of string tuples tells you what types should go in, and is more defensive against typos